### ISSUE TITLE
Check up for the new data

### ISSUE THREAD

**Issue by lizzieinvancouver** [lizzieinvancouver/ospree#355]
Following up from today, we want to cross-check our newly entered data for whether it makes sense, things to watch out for (it's all crops or the experiment seems odd) and whether it was all entered correctly. 

You are all in teams (see the CSV [here](https://github.com/lizzieinvancouver/ospree/tree/master/analyses/misc/datacheck). You are your teammate should (by next Wednesday):

- Check your papers and fill in the columns in CSV
- If you know you have an issue, make an issue for the paper (e.g., issue #354), fix it if you are sure you can, otherwise bring it to the group next Wednesday
- git push a new code file (ONE per group, named datacheckup_YOURTEAM.R) that shows what you did, makes plots, runs models.
- make a PDF presentation of no more than 6 slides showing your work, be sure to include at least 4 useful plots not in the current datacheckup_emw.R file and at least ONE figure showing the data and a model plotted on the data.

Please reply here if you have issues or to add things I said but am now forgetting! Thank you all!

**Comment by FaithJones** [comment]
Hey all. I am still not sure what the difference is between mall and mdbb in the cleaning script. What information should I be looking at to assess why I might have lost data?

**Comment by FaithJones** [comment]
Also Dan and I have been assigned ramos17, but I dont have a ramos17 in dbb. Did you mean ramos99?

**Comment by lizzieinvancouver** [comment]
@FaithJones mall is subsetting to your particular papers from ospree_clean.csv (ospree clean is the data run through https://github.com/lizzieinvancouver/ospree/blob/master/analyses/cleaning/cleanmerge_all.R (check out the README in the same folder). mdbb is cleaned further via https://github.com/lizzieinvancouver/ospree/blob/master/analyses/chilling/chillmerge_all.R (again, there's a README) and https://github.com/lizzieinvancouver/ospree/blob/master/analyses/bb_analysis/cleaning/bb_cleanmergeall.R. Here we subset to budburst data, so I think ramos17 is lost due to not having climate data. @cchambe12 might recall.

Also, I realize this is a lot to digest! @dbuona can hopefully help here. Mostly I tried to make the group's have an old hand (@cchambe12 @dbuona @AileneKane @MoralesCastilla)

**Comment by DeirdreLoughnan** [comment]
@lizzieinvancouver we think two of our papers are being removed because the forcing or photoperiod are entered as 'ambient.' Do you want us to try and get the climate data necessary to include these papers?

**Comment by AileneKane** [comment]
@lizzieinvancouver I have dug into the code for one of these two papers, fu18, and found that the problem is in "bb_analysis/cleaning/clean_ambientforcingfromdailyclimate.R" 
The chill.day file does not include fu18, nor does the the other climate (climdat) files, so perhaps the climate data have not been pulled for this site yet?
I also got an error when running this code, which may or may not be related: "Error in ans[ypos] <- rep(yes, length.out = len)[ypos] : replacement has length zero"

**Comment by lizzieinvancouver** [comment]
@DeirdreLoughnan @AileneKane Thanks guys! It does seem like E-OBS should have the data for fu18 (Antwerp) and we should check that error. If it's in the climate code, we'll need to ask @cchambe12 for help.

**Comment by AileneKane** [comment]
@lizzieinvancouver @DeirdreLoughnan @md-garner I have looked into a bit more and we definitely need to get insight from @cchambe12 because there is code written by her in bb_daily_dataprep_get_expclimdat.R that explicitly removes fu18 from the climate data code- not sure why!

**Comment by cchambe12** [comment]
Hi all, please keep making notes. I clearly need to revisit the climate data which I will fix once everyone updates their issues. I have a new plan of attack. Sorry for all of the frustration.

**Comment by lizzieinvancouver** [comment]
@cchambe12 Or I wonder if I am to blame here? I feel like there could have been a conversation where I suggested we skip fu18? (Maybe?) Anyway, we can discuss!

**Comment by cchambe12** [comment]
I've added a new column to the misc/datacheck/datacheck_assign.csv that labels why the chilling code or photoperiod conversions aren't working. We are missing a lot of field sample dates (richardson18 - because it is an open top chamber experiment). Fu18 is missing provenance information. Vitra17 is missing information in the chilltemp column. I think I need to rerun the chilling code for ramos17 and nanninga17 though

**Comment by lizzieinvancouver** [comment]
@FaithJones @AileneKane @DeirdreLoughnan @MoralesCastilla @dbuona @cchambe12 @md-garner Can you all add info on the three new columns in datacheck_assign.csv?

- location -- what general area was the study done?
- respvar -- what are the response variables(s)?
- sampling_yr -- what year(s) did the experiment happen?

Thank you!

**Comment by lizzieinvancouver** [comment]
@cchambe12 Do you recall what I need to check on man17?

**Comment by lizzieinvancouver** [comment]
@cchambe12 Ignore me, I remember!

**Comment by lizzieinvancouver** [comment]
@MoralesCastilla @legault I don't think we can enter the Fig 2A data in Prevey18 because there are no datapoints, just lines ... right? It would be really hard to decide how/where to click. Am I wrong?

**Comment by lizzieinvancouver** [comment]
@FaithJones @AileneKane @DeirdreLoughnan @MoralesCastilla @dbuona @cchambe12 @md-garner @DarwinSodhi Please do your new cleaning issues *before next Wednesday* ... note that only richardson18 and flynn18 did not have issues (if you cleaned those and think I am wrong, shout!), and @cchambe12 I think you're working on cleaning nanninga17 (if there are more issues there, let me know).

Note that the previous theme song for data cleaning was 'Are we out of the woods' (yes, by Taylor Swift, both @dbuona and I suggested it), we can have a new one now ...  anyone feeling Good as Hell? Other ideas?

**Comment by cchambe12** [comment]
nanninga17 should be all fixed! Two issues: 1) year was correct but field sample date was mis-entered 2) there seems to be random NAs in the new climate data for some studies. I have removed those rows for now (there were about 5-6 total rows across all studies and it seems to not be impacting our results but I do want to investigate this further to be sure)

**Comment by cchambe12** [comment]
Never mind! Climate data is all fixed!

**Comment by legault** [comment]
@lizzieinvancouver @MoralesCastilla 
prevey18 appears to have raw data available here: https://peerj.com/articles/5221/#supp-2

I believe we should be using these data instead of the summary statistics currently entered for prevey18

**Comment by lizzieinvancouver** [comment]
@cchambe12 will look at this and see if she can adapt Flynn code to extract what we need.

**Comment by cchambe12** [comment]
@lizzieinvancouver prevey18 raw data looks similar to what we already have inputted but it only has daystoflower as a respvar. Is it still worth scraping the raw data?

**Comment by lizzieinvancouver** [comment]
@lizzieinvancouver Tough call! In general we have always scraped all relevant respvars to keep our methods constant. If it is a ton of work to scrape it though, we could re-assess! Let me know.

**Comment by cchambe12** [comment]
@lizzieinvancouver Okay, I fixed prevey18. Since I was re-entering data, should I rewrite Mira's updated file ('data/update2019/ospree_2019update_my.csv), should I delete Mira's entries and then add the new prevey data to my update file or do something else entirely?

**Comment by cchambe12** [comment]
prevey18 is all set and updated!

**Comment by lizzieinvancouver** [comment]
@FaithJones @DeirdreLoughnan @AileneKane @cchambe12 @MoralesCastilla @DarwinSodhi @dbuona @legault @md-garner nalyses/misc/datacheck/datacheck_assign.csv is updated with new assigned paper (two set per team). Please wait fo @cchambe12  to give the go ahead before diving in on your new sets of data. 

Re-read the top of this issue for what to do (by Wed 29 April). You should fill in all columns *except* location, respvar, sampling_yr ... Also, I added a README to the folder.

Former teams: @cchambe12 and @DarwinSodhi, @FaithJones and @dbuona; @legault and @MoralesCastilla please fill in location, respvar, sampling_yr for your old papers.

**Comment by cchambe12** [comment]
@lizzieinvancouver @FaithJones @AileneKane @DarwinSodhi @dbuona @DeirdreLoughnan @md-garner @legault @MoralesCastilla  The data is all ready to be checked!

**Comment by lizzieinvancouver** [comment]
I checked a few issues from today:

* vitra17 -- we just made a call on this for converting data, it's all mapped out in issue #309 
* small amounts of missing data in richardson18 look okay to me so leaving this as is
* ramos17 -- has bb and percent flower, looks like we get the data we should so okay; I have a note to check dates ... I think it was just year (which we got from the authors, see issue #294 ) and it looked okay
* nanninga17 -- we generally enter data if it might be a duplicate but in this case I am 99.9% sure that the data in fig 2 are just a small subset of data (already scraped) in fig 1 so I think we're fine here.
* anzanello18 -- does not give the data in a way comparable to other studies (they use a complex f(x)) so I don't think we would ask for it
* malyshev18 -- is closed and a new issue set up to check its dates

So! I am closing this puppy. Thank you for all your efforts!
(See also commit b3aa7418291b317b58b4ae6427e3ac403715a22c)
